src/evalpoly.jl

```diff
@@ -0,0 +1,85 @@
+# matrix methods for evalpoly(X, p) = ∑ₖ Xᵏ⁻¹ p[k]
+
+# non-inplace fallback for evalpoly(X, p)
+function _evalpoly(X::AbstractMatrix, p)
+    Base.require_one_based_indexing(p)
+    p0 = isempty(p) ? Base.reduce_empty_iter(+, p) : p[end]
+    Xone = one(X)
+    S = Base.promote_op(*, typeof(Xone), typeof(Xone))(Xone) * p0
```

Comment:
Should this be
```suggestion
    S = convert(Base.promote_op(*, typeof(Xone), typeof(Xone)), Xone) * p0
```
to avoid unnecessary allocation of a new matrix in the likely event that `Xone isa Base.promote_op(*, typeof(Xone), typeof(Xone))`?

That aside, I'm not a fan of `promote_op` in general. What's the benefit here anyway?

Reply:
What would you use instead of `promote_op`?

Reply:
I'm missing background here to answer that. To start: what's wrong with just `S = Xone * p0`?

Reply:
I guess you want to make sure that a `Bool` eltype is promoted to `Int`:

```julia
julia> X = rand(Bool, 2, 2)
2×2 Matrix{Bool}:
 0  0
 0  1

julia> one(X)
2×2 Matrix{Bool}:
 1  0
 0  1

julia> one(X)^2
2×2 Matrix{Int64}:
 1  0
 0  1
```

And I agree about using `convert`.